### PR TITLE
op-node: Check withdrawals hash in P2P validation

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

Require P2P gossiped blocks to have empty withdrawals. Also clarify the specs around the handling of withdrawals in the op-node.

**Tests**

Unit tests for this new case

### CODE REVIEW

File: op-node/p2p/gossip.go [hunk 1]
```diff
@@ -325,6 +325,11 @@ func BuildBlocksValidator(log log.Logger, cfg *rollup.Config, runCfg GossipRunti
 			log.Warn("payload is on v2 topic, but does not have withdrawals", "bad_hash", payload.BlockHash.String())
 			return pubsub.ValidationReject
 		}
```
Comment (tynes) [inline op-node/p2p/gossip.go:327]: nit: all the others have a newline between them

File: op-node/p2p/gossip_test.go [hunk 2]
```diff
@@ -89,3 +99,65 @@ func TestVerifyBlockSignature(t *testing.T) {
...
+	// Params Set 1: Create the validation function
+	cfg := &rollup.Config{
+		L2ChainID: big.NewInt(100),
+	}
+	secrets, err := e2eutils.DefaultMnemonicConfig.Secrets()
+	require.NoError(t, err)
+	runCfg := &testutils.MockRuntimeConfig{P2PSeqAddress: crypto.PubkeyToAddress(secrets.SequencerP2P.PublicKey)}
+	signer := &PreparedSigner{Signer: NewLocalSigner(secrets.SequencerP2P)}
+
+	// valFnV1 := BuildBlocksValidator(testlog.Logger(t, log.LvlCrit), rollupCfg, runCfg, eth.BlockV1)
```
Comment (tynes) [inline op-node/p2p/gossip_test.go:133]: dead code

File: op-node/p2p/gossip_test.go [hunk 3]
```diff
@@ -89,3 +99,65 @@ func TestVerifyBlockSignature(t *testing.T) {
...
+	valFnV2 := BuildBlocksValidator(testlog.Logger(t, log.LvlCrit), cfg, runCfg, eth.BlockV2)
```
Comment (tynes) [inline op-node/p2p/gossip_test.go:134]: Its the `eth.BlockV2` that means canyon is active?

Comment (trianglesphere) [reply op-node/p2p/gossip_test.go:134]: Yes

File: op-node/p2p/gossip_test.go (hunk 2 above)
Comment (trianglesphere) [reply op-node/p2p/gossip_test.go:133]: yea. It's a todo to go expand the tests

**Review by mslipper — APPROVED** [review]

**Pull request merged**